src/pathwayvis/pathwayvis.component.ts

```diff
@@ -21,26 +21,54 @@ export class PathwayVisComponentController {
         this.$scope = $scope;
         this.mapOptions = mapOptions;
         this.showInfo = false;
+        this.$q = $q;
     }
 
-    public $onInit() {
-        this.mapOptions.init();
```

Comment:
So this was the call that emptied the datacards every time you navigated back to pathway visualizer. If this call isn't there, then the state persists between different dd-decaf modules.